advanced/learning_to_rank.rst

```diff
@@ -0,0 +1,142 @@
+
+LTR can be broadly seen in two stages: Learning the model & Ranking. Learning the model takes the training file as input and produces a model. After that given this learnt model, when a new query comes in, scores can be assigned to the documents associated to it.
+
+Preparing the Training file
+---------------------------
+
+Currently the ranking models supported by LTR are supervised learning models. A supervised learning model requires a labelled training data as an input. To learn a model using LTR you need to provide the training data in the following format.
+
+.. code-block:: none
+
+    0 qid:10032 1:0.130742 2:0.000000 3:0.333333 4:0.000000 ... 18:0.750000 19:1.000000 #docid = 1123323
+    1 qid:10032 1:0.593640 2:1.000000 3:0.000000 4:0.000000 ... 18:0.500000 19:0.023400 #docid = 4222333
+
+Here each row represents the document for the specified query. The first column is the relevance label and which can take non-negative values. The second column represents the queryid, and the last column is the docid. The third column represents the value of the features.
+
+As mentioned before, this process requires a training file in the above format. xapian-letor API empowers you to generate such training file. But for that you have to supply some information like:
+
+1. Query file: This file has information of queries to be involved in
+   learning and its id. It should be formatted in such a way::
+
+    2010001 'landslide,malaysia'
+    2010002 'search,engine'
+    2010003 'Monuments,of,India'
+    2010004 'Indian,food'
+
+   where 2010xxx being query-id followed by a comma separated query in
+   single-quotes.
+
+2. Qrel file: This is the file containing relevance judgements. It should
+   be formatted in this way::
+
+    2010003 Q0 19243417 1
+    2010003 Q0 3256433 1
+    2010003 Q0 275014 1
+    2010003 Q0 298021 0
+    2010003 Q0 1456811 0
+
+   where first column is query-id, third column is Document-id and fourth column being relevance label which is 0 for irrelevance and 1 for relevance. Second column is many times referred as 'iter' but doesn't really important for us.  All the fields are whitespace delimited. This is the standard format of almost all the relevance judgement files. If you have little different relevance judgement file then you can easily convert it in such file using basic 'awk' command.
```

Comment:
> If you have little different relevance judgement file then you can easily convert it in such file using basic 'awk' command.

Should be "convert it **to** such **a** file".

And "little different" sounds weird - I guess we're trying to say "If you have relevance judgements in a different format ..."

I'd also suggest thinking carefully about words like "simply", "easily", "trivial", etc in documentation.  Lana Brindley has a good talk about this I heard a few years back - I didn't find a video from a quick search, but this blog post covers some of the same ground:

http://lanabrindley.com/2011/11/20/open-source-docs-the-good-the-bad-and-the-wtf/

As a reader, if I don't see how to use "awk" to convert the format I have, then this sentence makes me feel stupid - this task is apparently "easy" but I've no idea how to do it, and now I'm stuck.  I may not know what "awk" is.

The word "easily" doesn't really add anything here.  Just dropping it is arguably an improvement, but we can also briefly explain what "awk" is - "... then you can convert it to this format using a text-processing tool such as 'awk'".  That's more helpful for a reader who hasn't heard of "awk" but has a favourite text-processing tool.